bin/postprocess_table.py

```diff
@@ -0,0 +1,296 @@
+#!/usr/bin/env python
+"""Post-process hits table so that it can be readily used for downloading genomes."""
+
+import argparse
+import logging
+import sys
+from pathlib import Path
+import pandas as pd
+import tempfile
+from subprocess import run, TimeoutExpired, CalledProcessError
+
+from tomlkit import boolean
+
+
+logger = logging.getLogger()
+
+
+def parse_args(argv=None):
+    """Define and immediately parse command line arguments."""
+    parser = argparse.ArgumentParser(
+        description="Postprocess hits table so that it can be readily used for downloading genomes",
+        epilog="Example: postprocess_filtered_table.py SRR12875570_pe-SRR12875570.tsv SRR12875570_pe-SRR12875570.postprocessed.tsv",
+    )
+    parser.add_argument(
+        "input",
+        metavar="INPUT",
+        type=Path,
+        help="Tsv file containing unprocessed hits table",
+    )
+    parser.add_argument(
+        "output",
+        metavar="OUTPUT",
+        type=Path,
+        help="Tsv file containing post-processed hits table",
+    )
+    parser.add_argument(
+        "-t",
+        "--ncbi-taxon-db",
+        metavar="Path",
+        type=Path,
+        default=Path("../temp/taxdump"),
+        help="Path to NCBI taxonomy database",
+    )
+    parser.add_argument(
+        "-l",
+        "--log-level",
+        help="The desired log level (default WARNING).",
+        choices=("CRITICAL", "ERROR", "WARNING", "INFO", "DEBUG"),
+        default="WARNING",
+    )
+    return parser.parse_args(argv)
+
+
+def get_unique_taxid_list(
+    df: pd.DataFrame, col_name: str = "taxid", classifier_name: str = ""
+) -> list[str]:
+    """Get a deduplicated list of taxids
+
+    Args:
+        df (pd.DataFrame): DataFrame where to extract the taxids
+        col_name (str, optional): Column name where the taxids are located. Defaults to "taxid".
+        classifier_name (str, optional): Name of the classifier which taxids we want to extract. If not given uses all classifiers. Defaults to "".
+
+    Returns:
+        list[str]: Deduplicated list of taxids of given classifier
+    """
+    if classifier_name:
+        taxid_col: pd.Series = df[col_name][df["classifier"] == classifier_name]
+    else:
+        taxid_col: pd.Series = df[col_name]
+    taxid_col.dropna(inplace=True)
+    return list(set(taxid_col.tolist()))
+
+
+def run_taxonkit_lineage(
+    temp_taxid_file: Path, db: Path, no_lineage: bool, show_name: bool
+) -> str:
+    """Run taxonkit lineage on a list of taxids file
+
+    Args:
+        temp_taxid_file (Path): Path to NamedTemporaryFile where taxids are each on one row
```

Comment:
`temp_taxid_file` is just a file that is temporarily created and it will be automatically deleted after the process exits. Why it's used here is because [`taxonkit lineage`](https://bioinf.shenwei.me/taxonkit/usage/#lineage) requires a text file with taxids one in each line as input and `tempfile.NamedTemporaryFile` is handy because it gets automatically deleted after it's been used. 

Yes, the taxids list comes from `get_unique_taxid_list()` and that list is used to create the contents to the `NamedTemporaryFile`.